src/traces/sankey/render.js

```diff
@@ -0,0 +1,568 @@
+        .ease(c.ease).duration(c.duration)
+        .style('transform', function(d) {return d.horizontal ? 'matrix(1,0,0,1,0,0)' : 'matrix(0,1,1,0,0,0)';});
+
+    var sankeyLink = sankeyLinks.selectAll('.sankeyLink')
+        .data(function(d) {
+            var uniqueKeys = {};
+            return d.sankey.links()
+                .filter(function(l) {return l.visible && l.value;})
+                .map(linkModel.bind(null, uniqueKeys, d));
+        }, keyFun);
+
+    sankeyLink.enter()
+        .append('path')
+        .classed('sankeyLink', true)
+        .call(attachPointerEvents, sankey, callbacks.linkEvents);
+
+    sankeyLink
+        .style('stroke', function(d) {return d.link.dy > 1 ? 'rgba(0,0,0,0)' : d.tinyColorHue;})
+        .style('stroke-width', function(d) {return d.link.dy > 1 ? 2 : 1;})
+        .style('stroke-opacity', function(d) {return d.tinyColorAlpha;})
+        .style('fill', function(d) {return d.tinyColorHue;})
+        .style('fill-opacity', function(d) {return d.tinyColorAlpha;});
+
+    sankeyLink.transition()
+        .ease(c.ease).duration(c.duration)
+        .attr('d', linkPath);
+
+    sankeyLink.exit().transition()
+        .ease(c.ease).duration(c.duration)
+        .style('opacity', 0)
+        .remove();
+
+    var sankeyNodeSet = sankey.selectAll('.sankeyNodeSet')
+        .data(repeat, keyFun);
+
+    sankeyNodeSet.enter()
+        .append('g')
+        .style('shape-rendering', 'geometricPrecision')
+        .classed('sankeyNodeSet', true);
+
+    sankeyNodeSet
+        .style('cursor', function(d) {
+            switch(d.arrangement) {
+                case 'fixed': return 'default';
+                case 'perpendicular': return 'ns-resize';
+                default: return 'move';
+            }
+        });
+
+    var sankeyNode = sankeyNodeSet.selectAll('.sankeyNode')
+        .data(function(d) {
+            var nodes = d.sankey.nodes();
+            var uniqueKeys = {};
+            persistOriginalPlace(nodes);
+            return nodes
+                .filter(function(n) {return n.visible && n.value;})
+                .map(nodeModel.bind(null, uniqueKeys, d));
+        }, keyFun);
+
+    sankeyNode.enter()
+        .append('g')
+        .classed('sankeyNode', true)
+        .call(updateNodePositions)
+        .call(attachPointerEvents, sankey, callbacks.nodeEvents);
+
+    sankeyNode
+        .call(attachDragHandler, sankeyLink, callbacks); // has to be here as it binds sankeyLink
+
+    sankeyNode.transition()
+        .ease(c.ease).duration(c.duration)
+        .call(updateNodePositions);
+
+    sankeyNode.exit().transition()
+        .ease(c.ease).duration(c.duration)
+        .style('opacity', 0)
+        .remove();
+
+    var nodeRect = sankeyNode.selectAll('.nodeRect')
+        .data(repeat);
+
+    nodeRect.enter()
+        .append('rect')
+        .classed('nodeRect', true)
+        .style('stroke-width', 0.5)
+        .call(Color.stroke, 'rgba(0, 0, 0, 1)')
```

Comment:
Can we make this color and width configurable? And ideally I guess the default color should be `Color.contrast(plot_bgcolor)`